src/common/meta/src/kv_backend/rds/postgres.rs

```diff
@@ -348,6 +392,163 @@ impl ExecutorFactory<PgClient> for PgExecutorFactory {
 /// It uses [deadpool_postgres::Pool] as the connection pool for [RdsStore].
 pub type PgStore = RdsStore<PgClient, PgExecutorFactory, PgSqlTemplateSet>;
 
+/// Creates a PostgreSQL TLS connector based on the provided configuration.
+///
+/// This function creates a rustls-based TLS connector for PostgreSQL connections,
+/// following PostgreSQL's TLS mode specifications exactly:
+///
+/// # TLS Modes (PostgreSQL Specification)
+///
+/// - `Disable`: No TLS connection attempted
+/// - `Prefer`: Try TLS first, fallback to plaintext if TLS fails (handled by connection logic)
+/// - `Require`: Only TLS connections, but NO certificate verification (accept any cert)
+/// - `VerifyCa`: TLS + verify certificate is signed by trusted CA (no hostname verification)
+/// - `VerifyFull`: TLS + verify CA + verify hostname matches certificate CN/SAN
+///
+/// (`VerifyCa` and `VerifyFull` are not implemented yet)
+///
+pub fn create_postgres_tls_connector(tls_config: &TlsOption) -> Result<MakeRustlsConnect> {
+    common_telemetry::info!(
+        "Creating PostgreSQL TLS connector with mode: {:?}",
+        tls_config.mode
+    );
+
+    // Create the TLS client configuration based on the mode and client cert requirements
+    let client_config = if !tls_config.cert_path.is_empty() && !tls_config.key_path.is_empty() {
+        // Client certificate authentication required
+        common_telemetry::info!("Loading client certificate for mutual TLS");
+        let cert_chain = load_certs(&tls_config.cert_path)?;
+        let private_key = load_private_key(&tls_config.key_path)?;
+
+        match tls_config.mode {
+            TlsMode::Disable => {
+                return PostgresTlsConfigSnafu {
+                    reason: "Cannot create TLS connector for Disable mode".to_string(),
+                }
+                .fail();
+            }
+            TlsMode::Prefer | TlsMode::Require => {
+                // For Prefer/Require: Accept any certificate (no verification)
+                let verifier = Arc::new(AcceptAnyVerifier);
+                ClientConfig::builder()
+                    .dangerous()
+                    .with_custom_certificate_verifier(verifier)
+                    .with_client_auth_cert(cert_chain, private_key)
```

Comment:
We will allow user to connect via tls without client certs. client certs are optional and for client authentication only. Actually there are 3 parts of tls:

- encryption
- client verifies server via ca
- server verifies client via client cert/key

If neither ca nor cleint cert provided, we still allow user to use tls for encryption.